bftengine/include/bftengine/ReservedPages.hpp

```diff
@@ -0,0 +1,116 @@
+// Concord
+//
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <typeindex>
+#include <algorithm>
+#include <tuple>
+#include <vector>
+#include <iostream>
+#include "demangle.hpp"
+
+namespace bftEngine {
+
+class ReservedPages {
+ public:
+  static uint32_t totalNumberOfPages() {
+    uint32_t numPages{0};
+    for (auto& elt : registry()) numPages += std::get<2>(elt);
+
+    return numPages;
+  }
+
+ protected:
+  // [idx, typeid, numberOfPages]
+  typedef std::vector<std::tuple<std::uint8_t, std::type_index, std::uint32_t>> Registry;
```

Comment:
I think wrapping this tuple in a type and then providing accesser methods will make the code much easier to read than using raw `std::get`s. With accessors, the user reading the code won't need to reference the type and then lookup what each index stands for, or attempt to keep mental track of that.